client/CPlayerInterface.cpp

```diff
@@ -1031,13 +1031,13 @@ void CPlayerInterface::yourTacticPhase(int distance)
 		boost::this_thread::sleep(boost::posix_time::millisec(1));
 }
 
-void CPlayerInterface::showComp(const Component &comp, std::string message)
+void CPlayerInterface::showComps(const std::pair<Component,Component> & comps, const std::pair<bool, bool> & use, std::string message)

```

Comment:
Use std::vector instead of std::pair

Reply:
@IvanSavenko why? I cannot show more than 2 objects now(

Reply:
You can show 0, 1 or 2 components now.
Vector would be more suitable instead of two interconnected pairs.

Besides - I don't think that server logic should be aware of how many objects can be shown in UI.
If you wish - drop "assert(size <= 2)" in UI loading.
Also, if somebody were to upgrade this code to support 3+ elements then this means rewriting entire code since you can't just store third element in pair.

If standard library had "vector with specific max capacity" class I would've used it. But it does not.

Reply:
@IvanSavenko will try to rewrite it using CComponentBox, maybe will work for 3+ components.

Reply:
Not sure if CComponentBox would change anything.
And I think we should use vector even with your current version.

Also, if you want to fit more elements - CComponent constructor accepts "size". You can use smaller image if you have over 2 components to display

Reply:
@IvanSavenko I do not want to write 8 constructors for each variant. I will try CComponentBox.

Reply:
@IvanSavenko stables is 0 components only if you do not have Cavaliers.

Reply:
Done with vector and CComponentBox. Is there any method than:
1. Gets only first row from incoming string
2. Gets required CTextBox height for showing string without scrollbar (in given width)?